### DISCUSSION TITLE
Vercel serverless function size limit exceed by reading image paths of local directories

### DISCUSSION THREAD

**Discussion by nidiotskr** [vercel/community#793] [Help] [answered]
![image](https://user-images.githubusercontent.com/100402861/186048889-fe54e450-7fab-4d78-b1f4-55119ecac40c.png)

Hello,

I'm getting an serverless function size limit error from reading image folders in public folder. I need to parse image paths from the local folder directory to make updating contents more easier.
![image](https://user-images.githubusercontent.com/100402861/186049009-7d4198fa-3348-4a51-b2ee-2697bfcf531b.png)
Please suggest a better way to solve this. Thank you.

**Comment by amyegan** [comment, marked as answer]
Hi @nidiotskr! Can you share your repo or a [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example)? If you can share that and a little more detail about what you're trying to do with your images, then I can try to help you come up with a solution.

**Reply by nidiotskr** [reply]
Hi @amyegan! Thanks for your support. I just changed my repository to public. This is really a simple static website, reading image paths from the folders in /public/assets and serve images as a gallery. please let me know if there is anything I can answer!

**Reply by amyegan** [reply]
Thanks for sharing your code @nidiotskr! It looks like the images are actually the problem. 😞 Your large image files are being pulled in as dependencies, and that is causing the Serverless Function to exceed the 50 MB size limit. You can get more details about large dependencies if you use [Vercel CLI](https://vercel.com/docs/cli) to run [`vercel build`](https://vercel.com/docs/cli#commands/build) on your local computer. Here's the warning I see:

```
Warning: Max serverless function size of 50 MB compressed or 250 MB uncompressed reached
Serverless Function's page: activities.js
Large Dependencies                  Uncompressed size  Compressed size
public/assets/imgs                             116 MB           116 MB
.next/server/pages                             1.4 MB           373 kB
node_modules/next/dist                        1.21 MB           295 kB
.next/server/font-manifest.json                693 kB           184 kB
node_modules/encoding/node_modules             329 kB           179 kB
node_modules/iconv-lite/encodings              303 kB           172 kB

All dependencies                               122 MB           118 MB
Max serverless function size was exceeded for 1 function
```

There are some options for getting around this limitation. One quick solution is to run a script on local whenever you add new events or event images. You could read the meta data and image paths on your local computer where the 50MB limit won't stop you. Then write the the data into a single JSON file, like `public/meta.json` and push it to your GitHub repo. And then you could read the meta data and image paths from that single JSON file when rendering the Activities page.

If you can tell me a bit more about your workflow, I can help you find a different solution that might work better for you.
